### ISSUE TITLE
Report undefined custom properties used in var() without fallback

### ISSUE THREAD

**Issue by schalkneethling** [schalkneethling/css-property-type-validator#54]
## Problem

When a declaration uses an undefined custom property through `var()` without a fallback, the validator currently does not report a diagnostic.

Example:

```css
.card {
  color: var(--brand-color);
}
```

If `--brand-color` is not available from the collected `@property` registry or otherwise known to the validator, and the `var()` call has no fallback, this should be reported as an unresolved variable reference.

## Current behavior

The validator skips or ignores these cases:

- Unregistered/unknown `var()` usages with no registered inputs produce no diagnostics.
- Mixed registered and unknown `var()` usages are skipped.

Relevant current behavior:

- `packages/core/src/validate.ts` ignores declarations where `registeredEntries.length === 0`.
- `packages/core/src/validate.ts` skips mixed registered and unregistered `var()` usages.
- `packages/core/test/validate-files.test.ts` has coverage expecting no diagnostic for unresolved/missing custom properties in mixed declarations.

## Expected behavior

For `var(--brand-color)` where `--brand-color` cannot be resolved and no fallback branch is provided, report a diagnostic such as:

> Custom property `--brand-color` is not defined and no fallback was provided.

For `var(--brand-color, red)` where `--brand-color` cannot be resolved, do not report the unresolved-variable diagnostic because the fallback can be used. The fallback should still be validated where the consuming property makes that possible.

## Possible implementation notes

- Add a new diagnostic reason/code, for example `unresolved-var-reference`.
- Detect unknown `var()` references where `getVarFallbackSource(varNode) === null`.
- Include `propertyName`, `expectedProperty`, `snippet`, and source location when available.
- Update tests that currently expect skipped/no diagnostic behavior for missing custom properties.
- Consider whether this should apply only to ordinary consuming declarations first, leaving assignment-site fallback behavior scoped separately if needed.

## Acceptance criteria

- `color: var(--brand-color);` reports an unresolved custom property diagnostic when `--brand-color` is not known.
- `color: var(--brand-color, red);` does not report unresolved custom property solely because `--brand-color` is unknown.
- Mixed declarations such as `border: var(--border-width) solid var(--missing-color);` report the missing no-fallback custom property instead of silently skipping.
- CLI, core formatter, VS Code diagnostics, and web output surface the new diagnostic consistently.
